include/vcpkg/base/fwd/expected.h

```diff
@@ -0,0 +1,16 @@
+#pragma once
+
+#include <string>
+#include <system_error>
+
```

Comment:
I think these largely defeat the purpose of a FWD header. I think most everywhere we use expected actually needs most of the things... unless we have a meaningful throughput win I would prefer to avoid adding such a header.

Reply:
There are many headers that declare a handful (or even one) function that returns `Expected`, but those functions are not necessarily called by consumers.

This fwd header specifically helps to avoid potential circular dependencies including `strings.h`, `stringliteral.h`, and `checks.h`.

Reply:
>There are many headers that declare a handful (or even one) function that returns Expected, but those functions are not necessarily called by consumers.

I don't think that necessarily matters. Forward declarations add complexity and create potential for 'spooky action at a distance' ABI issues; I think they need to demonstrate meaningful build throughput improvements before we deploy them.

>This fwd header specifically helps to avoid potential circular dependencies including strings.h, stringliteral.h, and checks.h.

In that case I think the change is broken and should have the layering violation fixed. There's no reason `api_stable_format` needs to be in such a low level header.

Reply:
Extracted `api_stable_format` out